### PR TITLE
Publish to maven central

### PR DESCRIPTION
**Pull request by galibey**
Add GitHub actions for publishing snapshots and releases to Maven repos.

Snapshots will be published after each merge to `main`.
Releases will be published by manual request.

Release java artifacts are compiled by Java 8 only. It is compatible with 8+ versions.

### CODE REVIEW

File: .github/workflows/push_snapshots.yml [hunk 1]
```diff
@@ -2,13 +2,42 @@ name: Push snapshot to Maven repo
...
+        with:
+          toolchain: stable
+          override: true
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          java-version: 8
+          distribution: 'adopt'
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
```
Comment (simlay) [inline .github/workflows/push_snapshots.yml:26]: Any specific reason to use the git hash rather than their released version?

**Review by simlay — APPROVED** [review]
All in all, I think this is good for now. I think running the build steps on each PR is worth it. I did this with the [Python publish CI](https://github.com/infinyon/fluvio-client-python/blob/bff5c1ad1809c415a87ab0ac18665ff0ba760f29/.github/workflows/publish.yml#L139) because the publish steps broke between publishes due to dependencies changing.

File: fluvio/build.gradle [hunk 2]
```diff
@@ -107,25 +100,66 @@ clean.dependsOn "clean-rust"
...
+                name = 'Fluvio Client for Java'
+                description = 'Java binding for Fluvio streaming platform'
+                url = 'https://github.com/infinyon/fluvio-client-java'
+                licenses {
+                    license {
+                        name = 'The Apache License, Version 2.0'
+                        url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                    }
+                }
+                developers {
```
[the remark this answers is not in the document]
Comment (galibey) [reply fluvio/build.gradle:138]: You're right, better to have only this contact. Changed.

File: .github/workflows/push_snapshots.yml (hunk 1 above)
Comment (galibey) [reply .github/workflows/push_snapshots.yml]: No reason actually. I changed to tags.

File: .github/workflows/do_release.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+on:
+  workflow_dispatch:
+    inputs:
+      release-version:
+        description: 'Version to be released (e.g. 0.12.13).'
+        required: true
+      next-snapshot-version:
+        description: 'Version to be set after the release (e.g. 0.12.14-SNAPSHOT).'
+        required: true
```
[the remark this answers is not in the document]
Comment (galibey) [reply .github/workflows/do_release.yml:11]: It is just easier and good enough for us at this moment. We will improve this later.

**Review by tjtelan — APPROVED** [review]
LGTM

**Pull request merged**